src/phoenix/server/api/input_types/InvocationParameters.py

```diff
@@ -1,20 +1,172 @@
-from typing import List, Optional
+from enum import Enum
+from typing import Any, List, Mapping, Optional, Union
 
 import strawberry
 from strawberry import UNSET
 from strawberry.scalars import JSON
 
 
+@strawberry.enum
+class CanonicalParameterName(str, Enum):
+    TEMPERATURE = "temperature"
+    MAX_COMPLETION_TOKENS = "max_completion_tokens"
+    STOP_SEQUENCES = "stop_sequences"
+    TOP_P = "top_p"
+    RANDOM_SEED = "random_seed"
+    TOOL_CHOICE = "tool_choice"
+
+
+@strawberry.enum
+class InvocationInputField(str, Enum):
+    value_int = "value_int"
+    value_float = "value_float"
+    value_bool = "value_bool"
+    value_string = "value_string"
+    value_json = "value_json"
+    value_string_list = "value_string_list"
+    value_boolean = "value_boolean"
+
+
 @strawberry.input
-class InvocationParameters:
-    """
-    Invocation parameters interface shared between different providers.
-    """
-
-    temperature: Optional[float] = UNSET
-    max_completion_tokens: Optional[int] = UNSET
-    max_tokens: Optional[int] = UNSET
-    top_p: Optional[float] = UNSET
-    stop: Optional[List[str]] = UNSET
-    seed: Optional[int] = UNSET
-    tool_choice: Optional[JSON] = UNSET
+class InvocationParameterInput:
+    invocation_name: str
+    canonical_name: Optional[CanonicalParameterName] = None
+    value_int: Optional[int] = UNSET
+    value_float: Optional[float] = UNSET
+    value_bool: Optional[bool] = UNSET
+    value_string: Optional[str] = UNSET
+    value_json: Optional[JSON] = UNSET
+    value_string_list: Optional[List[str]] = UNSET
+    value_boolean: Optional[bool] = UNSET
+
+
+@strawberry.interface
+class InvocationParameterBase:
+    invocation_name: str
+    canonical_name: Optional[CanonicalParameterName] = None
+    label: str
+    required: bool = False
+    hidden: bool = False
```

Comment:
required fields will need the user to provide some input, meaning we can provide a default value and the user needs to replace or edit it without deleting it in the UI

hidden fields will be populated by code at query time, and should not be displayed to the user. Think tool choice. That will be determined by UI elements outside of the invocation parameter form.

Reply:
I see. Is `tool_choice` the only example?

Reply:
Tool choice is not exactly hidden, is it? It's just not displayed in the invocation parameters form field. Kind of feels like UI logic leaking into the backend.

Reply:
yeah i agree here the hidden shouldn't leak into the api, we can choose to display things in different areas by name etc. in the ui

Reply:
I'm not sure I fully agree with this being bad "leakage" the point of this PR is to push decisions that used to be decided by the UI down into the backend to remove the coupling of the implementations

Reply:
That makes sense, in terms of defaults mins, maxes, supported parameters, etc. but why would the server decide what is hidden and not hidden to the user, like xander said tool choice is not hidden, it's just rendered in a different place once tools are added, that is a ui decision, not a server side one

Reply:
in a way it's because otherwise the UI needs to "know" about OpenAI or Anthropic models that make use of tool_choice being defined in that way—which is leakage in the reverse direction. I think tool choice being inferred by the UI at all is a type of leakage so there's going to be some intermingling no matter how we have it

Reply:
Yeah the ui has to understand providers and models either way. I'm unclear on what the hidden flag does to solve that issue. Happy to sync on it. My question would still be as an engineer adding a new invocation parameters to the backend, what is the decision making process to determine whether or not hidden should be true or false?

Reply:
it's roughly for parameters that might be useful for the UI to be aware of (especially if there is an underlying canonical version of them) that is less useful to expose directly in a playground

Reply:
After talking with Parker I am on team "remove hidden field".

I'd rather all fields have canonicalization and let the UI filter things out based on canonical names.